src/mavsdk/plugins/telemetry/include/plugins/telemetry/telemetry.h

```diff
@@ -1314,7 +1315,9 @@ class Telemetry : public PluginBase {
     /**
      * @brief Subscribe to 'status text' updates.
      */
-    void subscribe_status_text(StatusTextCallback callback);
+    Handle<StatusText> subscribe_status_text(StatusTextCallback&& callback);
```

Comment:
@JonasVautherin what do you think about using an r-value reference to pass the callback? I'm not too sure on whether to use l-value reference, r-value reference, or by value.

The r-value reference would allow the callback or `std::function` to be moved into the internal storage, however, it might be confusing for C++ users that have not encountered the r-value API before.
Also, moving the callbacks into the subscription means that you can't use the callback in another place which might make sense in most cases, however there might be cases where someone wants to re-use their callback.

Reply:
Nevermind, I switched to l-value reference: https://github.com/mavlink/MAVSDK/pull/1768/commits/a23cfddaded102aa7182c4411eac064ce08c305b